services/udp/udp.go

```diff
@@ -0,0 +1,131 @@
+package udp
+
+import (
+	"log"
+	"net"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/influxdb/influxdb/cluster"
+	"github.com/influxdb/influxdb/tsdb"
+)
+
+const (
+	udpBufferSize = 65536
+)
+
+type Server struct {
+	conn *net.UDPConn
+	addr *net.UDPAddr
+	wg   sync.WaitGroup
+
+	batchSize    int
+	batchTimeout time.Duration
+	database     string
+
+	PointsWriter interface {
+		WritePoints(p *cluster.WritePointsRequest) error
+	}
+
+	Logger *log.Logger
+}
+
+func NewServer(db string) *Server {
+	return &Server{
+		database: db,
+		Logger:   log.New(os.Stderr, "[udp] ", log.LstdFlags),
+	}
+}
+
+func (s *Server) SetBatchSize(sz int)             { s.batchSize = sz }
+func (s *Server) SetBatchTimeout(d time.Duration) { s.batchTimeout = d }
+
+func (s *Server) ListenAndServe(iface string) (err error) {
+	s.addr, err = net.ResolveUDPAddr("udp", iface)
+	if err != nil {
+		s.Logger.Printf("Failed to resolve UDP address %s: %s", iface, err)
+		return err
+	}
+
+	s.conn, err = net.ListenUDP("udp", s.addr)
+	if err != nil {
+		s.Logger.Printf("Failed to set up UDP listener at address %s: %s", s.addr, err)
+		return err
+	}
+
+	s.Logger.Printf("Started listening on %s", iface)
+
+	batcher := tsdb.NewPointBatcher(s.batchSize, s.batchTimeout)
+	batcher.Start()
+
+	// Start processing batches.
+	var wg sync.WaitGroup
+	done := make(chan struct{})
```

Comment:
This needs to be part of the server struct. Also, I don't see us actually `close` the channel, which needs to be done in `server.Close`